ethcore/src/verification/verification.rs

```diff
@@ -308,10 +305,13 @@ pub fn verify_header_params(header: &Header, engine: &EthEngine, is_full: bool,
 	if is_full {
 		const ACCEPTABLE_DRIFT: Duration = Duration::from_secs(15);
 		// this will resist overflow until `year 2037`
-		let max_time = SystemTime::now() + ACCEPTABLE_DRIFT;
-		let invalid_threshold = max_time + ACCEPTABLE_DRIFT * 9;
-		let timestamp = UNIX_EPOCH.checked_add(Duration::from_secs(header.timestamp()))
-			.ok_or(BlockError::TimestampOverflow)?;
+		let now = SystemTime::now();
+		let now = now.duration_since(UNIX_EPOCH).expect("now is after UNIX_EPOCH; qed");
+		let max_time = now.checked_add(ACCEPTABLE_DRIFT)
+			.unwrap_or(Duration::from_secs(u64::max_value()));
+		let invalid_threshold = max_time.checked_add(ACCEPTABLE_DRIFT * 9)
+			.unwrap_or(Duration::from_secs(u64::max_value()));
```

Comment:
This is not the same behavior as before is it? It used to error out and now it just silently proceeds with `u64::max_value()`, is that an issue?

Reply:
Yeah, I think the only change is that if `timestamp == u64::max_val() && max_time == u64::max_val() && invalid_threshold == u64::max_val() ` then we won't detect an error. 

However, before we returned `TimestampOverflow` when timestamp >  `i32::max_value`

Also, it should not be possible to force `SystemClock` to u64::max_value because it is represented i32/i64 on the systems I know of

Might be better to keep `max_time` and `invalid_threshold` as Options and return some other error when both are None?

Reply:
As long as it doesn't introduce any consensus-relevant difference, e.g. an old chain with a block with a forged timestamp used to be invalid there and is valid now (or vice versa) so that there could be a split depending on if you reject or if you cap out.

Reply:
No, as long as we are not near `2038` it should be safe :)